### ISSUE TITLE
MlpPolicy: expose encoder/feature tap for custom auxiliary losses

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/thrust#1]
## Motivation

Several MARL methods (PMIC, our Slepian-Wolf specialization regularizer, behavioural-diversity bonuses, etc.) need to compute losses on the *shared trunk* output, not just the final action logits/value. Today the trunk activations are private:

\`\`\`rust
// src/policy/mlp.rs
pub struct MlpPolicy {
    vs: nn::VarStore,
    shared: nn::Sequential,     // <-- private
    policy_head: nn::Linear,
    value_head: nn::Linear,
    ...
}

pub fn forward(&self, obs: &Tensor) -> (Tensor, Tensor) {
    let features = self.shared.forward(obs);   // <-- discarded after heads
    let logits = self.policy_head.forward(&features);
    let values = self.value_head.forward(&features).squeeze_dim(-1);
    (logits, values)
}
\`\`\`

## Proposal

Add a public method returning the shared-trunk activations:

\`\`\`rust
/// Shared-trunk features, suitable for auxiliary losses (representational
/// diversity / redundancy / contrastive penalties). Gradient flows back
/// through the trunk.
pub fn encoder_features(&self, obs: &Tensor) -> Tensor {
    self.shared.forward(obs)
}
\`\`\`

Optionally extend \`forward\` to a 3-tuple \`(logits, values, features)\` and have \`forward_2\` keep the current signature for backwards compatibility.

## Use case

In the Slepian-Wolf MARL specialization experiment (P3) we run \`N\` policies on a synchronized batch and need \`encoder_features(obs)\` for each one to compute a pairwise cross-correlation penalty. Currently we mirror the Python pattern of reaching into a private field, which won't work in Rust.

## Related

Pairs with #2 (auxiliary-loss hook in \`PPOTrainer\`) and #5 (joint synchronized trainer).

**Comment by rjwalters** [comment]
## Curator audit: already implemented on `main`

Audited the codebase against the proposal in this issue and the work is fully delivered. Recommending **close as completed**.

### What landed

Commit [`049bbc4`](https://github.com/rjwalters/thrust/commit/049bbc4) ("policy: expose MlpPolicy::encoder_features tap") added the exact API this issue requested:

- `src/policy/mlp.rs:192` — `pub fn encoder_features(&self, obs: &Tensor) -> Tensor` returning the shared-trunk activations of shape `[batch, hidden_dim]`. Doc comment explicitly calls out the auxiliary-loss / cross-agent-redundancy / behavioural-diversity use case.
- `src/policy/mlp.rs:174` — `forward()` left at the original `(logits, values)` 2-tuple signature, preserving backwards compatibility. The "optional 3-tuple" alternative from the proposal was correctly *not* taken — a separate tap is cleaner and the implementation hasn't suffered for it.

### What the proposal asked for vs. what shipped

| Proposed | Shipped | Note |
|---|---|---|
| `pub fn encoder_features(&self, obs) -> Tensor` | `src/policy/mlp.rs:192` |  Exact match |
| Gradient flows back through trunk | Verified by `tests/test_encoder_features.rs::encoder_features_carries_gradient` and `..._and_forward_use_same_trunk` |  Verified |
| Pairs with #2 (PPO auxiliary-loss hook) | `505e7f1` shipped #2; `tests/test_ppo_aux_loss.rs` exercises `encoder_features` through the PPO aux hook |  Pair landed |
| Pairs with #5 (joint synchronized trainer) | `40ec676` shipped a joint trainer example; #5 still open |  Issue #5 remains the right tracking issue for that work |

### Additional related deliveries (bonus scope, not requested here)

The same surface was added to `MultiDiscreteMlpPolicy` (`src/policy/multi_discrete_mlp.rs:178`), which is the right thing to do given #3/#5 use it. Tested in `tests/test_multi_discrete_mlp.rs:81`.

### Real downstream consumers

`encoder_features` is now used by:
- `examples/games/bucket_brigade/train_p3.rs:345` — the Slepian-Wolf P3 specialization regularizer, which is the motivating use case named in this issue's "Use case" section.
- `tests/test_ppo_aux_loss.rs` — exercising the #2 aux-loss hook end-to-end.
- `src/train/ppo/trainer.rs:140` — docstring reference pointing future users at the right tap.

### Caveat noted in the commit message

The commit places tests in `tests/` (integration) rather than `#[cfg(test)] mod tests` (in-module) because the lib unit-test build is currently broken on `main`. That is **tracked separately as #7** ("Lib unit tests broken on main"). Moving these encoder tests in-module is a one-line follow-up once #7 lands, and does not warrant keeping this issue open.

### Recommendation

Close as completed. No follow-up work in scope for this issue. The follow-ups that *do* exist are already tracked:
- #2 (aux loss hook) — done (`505e7f1`), can also be closed if not already
- #5 (joint sync trainer) — open, separate scope
- #7 (lib unit tests on main) — open, would let `test_encoder_features.rs` move in-module

**Comment by rjwalters** [comment]
Closing as already implemented on `main` — see curator audit above. Tap landed in 049bbc4, downstream consumers exist, follow-ups are tracked in #5 and #7.
